apollo-biz/src/main/java/com/ctrip/framework/apollo/biz/service/ItemService.java

```diff
@@ -200,6 +210,26 @@ public Item update(Item item) {
 
   private boolean checkItemValueLength(long namespaceId, String value) {
     int limit = getItemValueLengthLimit(namespaceId);
+    Optional<Namespace> currentNamespace = namespaceRepository.findById(namespaceId);
+    if(currentNamespace.isPresent()) {
+      String appId = currentNamespace.get().getAppId();
+      String namespaceName = currentNamespace.get().getNamespaceName();
+      String clusterName = currentNamespace.get().getClusterName();
+      if(appId != null && namespaceName != null && clusterName != null) {
+        Matcher m = clusterPattern.matcher(clusterName);
+        boolean isGray = m.matches();
+        if (isGray) {
+          Cluster currentCluster = clusterRepository.findByAppIdAndName(appId, clusterName);
```

Comment:
why not use `NamespaceService#findParentNamespace(com.ctrip.framework.apollo.biz.entity.Namespace)`?